src/CDN/CloudFront.php

```diff
@@ -152,83 +172,77 @@ public function flush($string)
-            ],
-            'CallerReference' => $this->getCallerReference($normalizedPaths),
-        ];
-
-        $arguments = [
-            'DistributionId' => $this->distributionId,
-        ];
-
-        // @todo: Remove the following check and the `else` block when support for aws/aws-sdk-php < 3.0 is dropped.
-        if (class_exists(Sdk::class)) {
-            // AWS v3.x.
-            $arguments += ['InvalidationBatch' => $invalidationBatch];
-        } else {
-            // AWS v2.x.
-            $arguments += $invalidationBatch;
-        }
-
         try {
-            $result = $this->getClient()->createInvalidation($arguments);
+            $result = $this->client->createInvalidation([
+                'DistributionId' => $this->distributionId,
+                'Paths' => [
+                    'Quantity' => \count($normalizedPaths),
+                    'Items' => $normalizedPaths,
+                ],
+                'CallerReference' => $this->getCallerReference($normalizedPaths),
+            ]);
 
-            if (!\in_array($status = $result->get('Status'), ['Completed', 'InProgress'], true)) {
-                throw new \RuntimeException('Unable to flush : '.$status);
+            $status = $result->get('Status');
+
+            if (false === array_search($status, self::AVAILABLE_STATUSES, true)) {
+                throw new \RuntimeException(sprintf('Unable to determine the flush status from the given response: "%s".', $status));
             }
 
             return $result->get('Id');
         } catch (CloudFrontException $ex) {
-            throw new \RuntimeException('Unable to flush : '.$ex->getMessage());
-        }
-    }
-
-    /**
-     * For testing only.
-     *
-     * @param CloudFrontClient $client
-     */
-    public function setClient($client)
-    {
-        if (!$client instanceof CloudFrontClient) {
-            @trigger_error('The '.__METHOD__.' expects a CloudFrontClient as parameter.', E_USER_DEPRECATED);
+            throw new \RuntimeException('Unable to flush: '.$ex->getMessage(), $ex->getCode(), $ex);
         }
-
-        $this->client = $client;
     }
 
     public function getFlushStatus($identifier)
     {
         try {
-            $result = $this->getClient()->getInvalidation([
+            $result = $this->client->getInvalidation([
                 'DistributionId' => $this->distributionId,
                 'Id' => $identifier,
             ]);
 
-            return array_search($result->get('Status'), self::getStatusList(), true);
+            $status = array_search($result->get('Status'), self::AVAILABLE_STATUSES, true);
+
+            if (false === $status) {
+                @trigger_error(sprintf(
+                    'Returning a value not present in the `%s::STATUS_*` constants is deprecated since sonata-project/media-bundle 3.x'
+                    .' and will not be possible in version 4.0.',
+                    CDNInterface::class
```

Comment:
```suggestion
                    'Returning a value not present in the `%s::STATUS_*` constants from %s() is deprecated since sonata-project/media-bundle 3.x'
                    .' and will not be possible in version 4.0.',
                    CDNInterface::class,
                    __METHOD__
```

Reply:
ping @phansys

Reply:
Thank you! The suggestion was committed.